src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+  hooks.stateReducers.push(reducer);
+  hooks.useInstanceBeforeDimensions.push(useInstanceBeforeDimensions);
+};
+
+const defaultGetResizerProps = (ownerDocument: Document | undefined) => (
+  props: TableKeyedProps,
+  {
+    instance,
+    header,
+    nextHeader,
+  }: {
+    instance: TableInstance;
+    header: HeaderGroup;
+    nextHeader: HeaderGroup;
+  },
+) => {
+  if (!ownerDocument) {
+    return props;
+  }
+
+  const { dispatch, flatHeaders } = instance;
+
+  const onResizeStart = (
+    e: React.TouchEvent & React.MouseEvent,
+    header: HeaderGroup,
+  ) => {
+    let isTouchEvent = false;
+    if (e.type === 'touchstart') {
+      // lets not respond to multiple touches (e.g. 2 or 3 fingers)
+      if (e.touches && e.touches.length > 1) {
+        return;
+      }
+      isTouchEvent = true;
+    }
+
+    // Setting `width` here because it might take several rerenders until actual column width is set.
+    flatHeaders.forEach((h) => {
+      if (!h.width) {
+        h.width = h.resizeWidth;
+      }
+    });
+
+    const headerIdWidths = getLeafHeaders(header).map((d) => [d.id, d.width]);
+    const nextHeaderIdWidths = nextHeader
+      ? getLeafHeaders(nextHeader).map((d) => [d.id, d.width])
+      : [];
+
+    const clientX = isTouchEvent ? Math.round(e.touches[0].clientX) : e.clientX;
+
+    const dispatchMove = (clientXPos: number) =>
+      dispatch({ type: actions.columnResizing, clientX: clientXPos });
+    const dispatchEnd = () =>
+      dispatch({
+        type: actions.columnDoneResizing,
+      });
+
+    const handlersAndEvents = {
+      mouse: {
+        moveEvent: 'mousemove',
+        moveHandler: (e: MouseEvent) => dispatchMove(e.clientX),
+        upEvent: 'mouseup',
+        upHandler: () => {
+          ownerDocument.removeEventListener(
+            'mousemove',
+            handlersAndEvents.mouse.moveHandler,
+          );
+          ownerDocument.removeEventListener(
+            'mouseup',
+            handlersAndEvents.mouse.upHandler,
+          );
+          dispatchEnd();
+        },
+      },
+      touch: {
+        moveEvent: 'touchmove',
+        moveHandler: (e: TouchEvent) => {
+          if (e.cancelable) {
+            e.preventDefault();
+            e.stopPropagation();
+          }
+          dispatchMove(e.touches[0].clientX);
+          return false;
+        },
+        upEvent: 'touchend',
+        upHandler: () => {
+          ownerDocument.removeEventListener(
+            handlersAndEvents.touch.moveEvent,
+            handlersAndEvents.touch.moveHandler,
+          );
+          ownerDocument.removeEventListener(
+            handlersAndEvents.touch.upEvent,
+            handlersAndEvents.touch.moveHandler,
+          );
+          dispatchEnd();
+        },
+      },
+    };
+
+    const events = isTouchEvent
+      ? handlersAndEvents.touch
+      : handlersAndEvents.mouse;
+    const passiveIfSupported = passiveEventSupported()
+      ? { passive: false }
```

Comment:
If we are just setting passive to false, then do we really need the utility? Isn't false default?

Reply:
> some browsers (specifically, Chrome and Firefox) have changed the default value of the passive option to true for the touchstart and touchmove events on the document-level nodes Window, Document, and Document.body
I guess this is for the touch events also found that `d3-drag` lib also says that passive event is not appropriate for the drag: https://github.com/d3/d3-drag/issues/35#issuecomment-314798648